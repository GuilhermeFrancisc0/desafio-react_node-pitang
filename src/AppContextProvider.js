import React, { createContext, useState, useEffect } from 'react';
// import { toast } from 'react-toastify';
// import axios from './utils/api';

export const AppContext = createContext(); // Criação no app para ser "escutada" nas pages

export default function AppContextProvider({ children }) {
  // Criação do state para renderizar no frontend
  const [agendamentos, setAgendamentos] = useState([]);

  const fetchData = async () => { // Busca os dados agendamentos do backend
    // Comentado até finalizar o backend
    // try {
    //   const response = await axios.get('/agendamento'); // get para buscar na rota de agendamento
    //   setAgendamentos(response.data.data); // atualizar a lista de agendamentos local
    // } catch (error) {
    //   toast.error(error.message);
    // }
  };

  useEffect(() => { // Começo do código, quando a página é renderizada
    fetchData(); // Ativa a função de buscar dados
  }, []);
  return (
    <AppContext.Provider value={[agendamentos, setAgendamentos, fetchData]}>
      {children}
    </AppContext.Provider>
  );
}
