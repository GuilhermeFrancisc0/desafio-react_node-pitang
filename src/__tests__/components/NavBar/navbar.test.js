/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../../../components/Navbar';
import AppContextProvider from '../../../AppContextProvider';

const NavBarWrapper = () => (
  <AppContextProvider>
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>
  </AppContextProvider>
);

test('NavBar renders', () => {
  const { asFragment } = render(<NavBarWrapper />);

  expect(asFragment()).toMatchSnapshot();
});
