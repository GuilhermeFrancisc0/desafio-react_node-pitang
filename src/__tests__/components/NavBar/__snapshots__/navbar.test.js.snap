// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Portal vacina COVID-19
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      />
    </div>
  </nav>
</DocumentFragment>
`;
